base/deprecated.jl

```diff
@@ -634,3 +634,15 @@ end
 
 @deprecate mmap_bitarray{N}(::Type{Bool}, dims::NTuple{N,Integer}, s::IOStream, offset::FileOffset=position(s)) mmap(s, BitArray, dims, offset)
 @deprecate mmap_bitarray{N}(dims::NTuple{N,Integer}, s::IOStream, offset=position(s)) mmap(s, BitArray, dims, offset)
+
+function require(mod::AbstractString)
+    depwarn("`require` is deprecated, use `using` or `import` instead", :require)
+    Base.require(symbol(mod))
```

Comment:
should this attempt to take the `basename(mod)` and remove the potential `.jl` suffix? or should it be a copy of the old `Base.require(::AbstractString)` logic?

also, `Base.` is unnecessary.